javascript/ql/lib/semmle/javascript/AMD.qll

```diff
@@ -91,7 +91,9 @@ class AmdModuleDefinition extends CallExpr instanceof AmdModuleDefinition::Range
   Function getFactoryFunction() { TValueNode(result) = this.getFactoryNodeInternal() }
 
   private EarlyStageNode getFactoryNodeInternal() {
-    result = TValueNode(this.getLastArgument())
+    exists(Function factoryFunction | factoryFunction = this.getArgument(_) |
+      result = TValueNode(factoryFunction)
+    )
```

Comment:
The restriction to `Function` here is too restrictive as we may need to follow some local flow steps before we get to the function (see the recursive case below). Some tests are failing because of this. Could you try simply changing `getLastArgument()` to `getArgument(_)`?

Reply:
I've changed the construction of `TValueNode` to `TValueNode(this.getArgument(_))`.